File: app/src/main/java/com/romellfudi/callbacks/MathCallback.java
package com.romellfudi.callbacks;

/**
 * Created by romelldominguez on 9/19/16.
 */
public class MathCallback {

    Callback callback;

    void Mathsuccessed(Object... objects){
        callback.successed(objects);
    }

    void Mathfailed(NumberFormatException exception,Object... objects){
        callback.failed(exception.getMessage(),objects);
        ;
    }



}

File: app-kotlin/src/main/java/com/romellfudi/callbacks/MathCallback.kt
package com.romellfudi.callbacks

/**
 * Created by romelldominguez on 9/19/16.
 */
open class MathCallback {
    var callback: Callback? = null
    open fun Mathsuccessed(vararg objects: Any) {
        callback!!.successed(*objects)
    }

    open fun Mathfailed(exception: NumberFormatException, vararg objects: Any?) {
        callback!!.failed(exception.message!!, objects)
    }
}